tidy3d/plugins/design/method.py

```diff
@@ -23,150 +21,87 @@ class Method(Tidy3dBaseModel, ABC):
     name: str = pd.Field(None, title="Name", description="Optional name for the sweep method.")
 
     @abstractmethod
-    def run(self, parameters: Tuple[ParameterType, ...], fn: Callable) -> Tuple[Any]:
+    def run(self, parameters: Tuple[ParameterType, ...], run_fn: Callable) -> Tuple[Any]:
         """Defines the search algorithm (sequential)."""
 
-    @abstractmethod
-    def sample(self, parameters: Tuple[ParameterType, ...], **kwargs) -> Dict[str, Any]:
-        """Defines how the design parameters are sampled."""
+    def _force_int(self, next_point: dict, parameters: list):
+        """Convert a float asigned to an int parameter to be an int. Update dict in place."""
 
-    @staticmethod
-    def assert_hashable(fn_args: dict) -> None:
-        """Raise error if the function arguments aren't hashable (do before computation)."""
-        fn_args_tuple = tuple(fn_args)
-        try:
-            hash(fn_args_tuple)
-        except TypeError:
-            raise ValueError(
-                "Function arguments must be hashable. "
-                "Parameter sweep tool won't work with sets of lists, dicts or numpy arrays. "
-                "Convert these to 'tuple' for a workaround."
-            )
+        for param in parameters:
+            if isinstance(param, ParameterInt):
+                # Using int(round()) instead of just int as int always rounds down making upper bound value impossible
+                next_point[param.name] = int(round(next_point[param.name], 0))
 
+    @staticmethod
+    def _extract_output(output: list, sampler: bool = False) -> Tuple:
+        """Format the user function output for further optimization and result storage."""
+        if sampler:
+            return output
+
+        if all(isinstance(val, (float, int)) for val in output):
+            # No aux_out
+            return (output, None)
+
+        if all(isinstance(val, (list, Tuple)) for val in output):
+            if all(isinstance(val[0], (float, int)) for val in output):
+                float_out = []
+                aux_out = []
+                for val in output:
+                    float_out.append(val[0])
+                    aux_out.append(val[1])
```

Comment:
should this error if not all `len(val) == 2`?

Reply:
Good point, it's now raising something here.